backend/api/management/commands/generate_fixtures.py

```diff
@@ -1,218 +1,297 @@
-from django.core.files.base import ContentFile as _
+import structlog
+from django.core.files.base import ContentFile
 from django.core.management.base import BaseCommand
 
 from api.models import ComputePlan
 from api.models import ComputeTask
 from api.tests import asset_factory as factory
 
+logger = structlog.get_logger(__name__)
+
 
 class Command(BaseCommand):
     help = "Generate fixtures"
 
     def handle(self, *args, **options):
-        simple_algo = factory.create_algo(
+        logger.debug("Generate test data")
+        self.create_algos()
+        self.create_data_manager()
+        self.create_data_samples()
+        self.create_empty_cp()
+        for status in [
+            ComputePlan.Status.PLAN_STATUS_TODO,
+            ComputePlan.Status.PLAN_STATUS_DOING,
+            ComputePlan.Status.PLAN_STATUS_DONE,
+            ComputePlan.Status.PLAN_STATUS_FAILED,
+            ComputePlan.Status.PLAN_STATUS_CANCELED,
+        ]:
+            self.create_basic_cp(status)
+        self.create_aggregate_cp()
+        self.create_composite_cp()
+
+    def create_algos(self):
+        logger.debug("  Create algos")
+        self.simple_algo = factory.create_algo(
             inputs=factory.build_algo_inputs(["datasamples", "opener", "model"]),
             outputs=factory.build_algo_outputs(["model"]),
-            name="simple algo",
+            name="simple",
         )
-        aggregate_algo = factory.create_algo(
+        factory.create_algo_files(
+            key=self.simple_algo.key,
+            description=ContentFile("Simple algo"),
+        )
+
+        self.aggregate_algo = factory.create_algo(
             inputs=factory.build_algo_inputs(["model"]),
```

Comment:
I'm not sure this would be a realistic input for an aggregate algo as `multiple=False` for `"model"` but maybe that's going too much into details or I am missing something.

Reply:
Yes it would be better, thanks.

Reply:
Done in this commit https://github.com/Substra/substra-backend/pull/495/commits/5ae663155817196cd4101147d73d235b39e9f89e